scripts/ci/validate-metadata-files.js

```
@@ -0,0 +1,166 @@
+	const packagesWithMetadata = packagesData.filter( p => p.metadata );
+
+	const invalidMetadata = [];
+
+	for ( const packageData of packagesWithMetadata ) {
+		const missingExports = getMissingExports( packageData );
+		const missingIcons = getMissingIcons( packageData, options );
+
+		if ( ![ ...missingExports, ...missingIcons ].length ) {
+			continue;
+		}
+
+		invalidMetadata.push( {
+			name: packageData.name,
+			missingExports,
+			missingIcons
+		} );
+	}
+
+	if ( !invalidMetadata.length ) {
+		console.log( green( 'Validation successful.' ) );
+
+		return;
+	}
+
+	for ( const x of invalidMetadata ) {
+		console.log( red( `\nPackage "${ x.name }" has invalid metadata entries:` ) );
+
+		if ( x.missingExports.length ) {
+			console.log( red( [
+				'Missing exports:',
+				...x.missingExports.map( missingExport => ` - ${ missingExport }` )
+			].join( '\n' ) ) );
+		}
+
+		if ( x.missingIcons.length ) {
+			console.log( red( [
+				'Missing icons:',
+				...x.missingIcons.map( missingIcon => ` - ${ missingIcon }` )
+			].join( '\n' ) ) );
+		}
+	}
+
+	process.exit( 1 );
+}
+
+/**
+ * Gets plugins from metadata file at provided path.
+ * Additionally, adds `packageName` of the plugin to returned data.
+ */
+async function getPackageData( packagePath ) {
+	const name = packagePath.split( '/' ).at( -1 );
+
+	const pkgJsonPath = upath.join( packagePath, 'package.json' );
+	const pkgJsonContent = await fs.readJSON( pkgJsonPath );
+
+	const metadataPath = upath.join( packagePath, 'ckeditor5-metadata.json' );
+	const indexPath = upath.join( packagePath, pkgJsonContent.main );
+
+	const metadata = await fs.exists( metadataPath ) ? await fs.readJSON( metadataPath ) : null;
+	const index = await fs.exists( indexPath ) ? await fs.readFile( indexPath, 'utf-8' ) : null;
+
+	return { name, metadata, index };
+}
+
+function getMissingExports( packageData ) {
+	if ( !packageData.index ) {
+		throw new Error( 'missing index' );
+	}
```

Do we want to throw an error from this script? It would be better to collect packages without index file and print error message at the end.